Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -419,10 +433,14 @@ def fetch_events_command(
                 config_id = event.get('attackData', {}).get('configId', "")
                 policy_id = event.get('attackData', {}).get('policyId', "")
                 demisto.debug(f"Couldn't decode event with {config_id=} and {policy_id=}, reason: {e}")
-        total_events_count += len(events)
-        demisto.info(f"Got {len(events)} events, and {offset=}")
-        yield events, offset, total_events_count
-    yield [], offset, total_events_count
+        demisto.info("Preparing to deduplicate events, currently got {len(events)} events.")
+        deduped_events = [event for event, hashed_event_from_current_run in zip(
+            events, hashed_events_from_current_run) if hashed_event_from_current_run not in hashed_events_from_previous_run]
```

Comment:
this is gonna be order n^2 with many events that we know akamai has will be slow
lets do some set operation, basically current_run - previous_run. We can keep the events itself in a map tied to the hashes as ids. let me know if you want to discuss

Reply:
NP